:::NVLOGv0.1.0 ncf 1603861909.683275938 (ncf.py:171) cpu_info: {"num": 48, "name": "AMD Ryzen Threadripper 3960X 24-Core Processor"}
:::NVLOGv0.1.0 ncf 1603861909.689881563 (ncf.py:171) mem_info: {"ram": "125G"}
:::NVLOGv0.1.0 ncf 1603861909.923778296 (ncf.py:171) gpu_info: {"driver_version": "455.28", "num": 2, "name": ["GeForce RTX 3080", "GeForce RTX 3080"], "mem": ["10015 MiB", "10018 MiB"]}
:::NVLOGv0.1.0 ncf 1603861909.926249504 (ncf.py:174) args: {"data": "/data/ncf/cache/ml-20m", "epochs": 2, "batch_size": 1200000, "valid_batch_size": 1048576, "factors": 64, "layers": [256, 256, 128, 64], "negative_samples": 4, "learning_rate": 0.0045, "topk": 10, "seed": 1, "threshold": 1.0, "beta1": 0.25, "beta2": 0.5, "eps": 1e-08, "dropout": 0.5, "checkpoint_dir": "/data/checkpoints/", "load_checkpoint_path": null, "mode": "train", "grads_accumulated": 1, "opt_level": "O2", "local_rank": 0, "distributed": false, "world_size": 1}
Saving results to /data/checkpoints/
:::NVLOGv0.1.0 ncf 1603861909.927096605 (ncf.py:184) preproc_hp_sample_eval_replacement: true
:::NVLOGv0.1.0 ncf 1603861909.927508831 (ncf.py:185) input_hp_sample_train_replacement: true
:::NVLOGv0.1.0 ncf 1603861909.927903175 (ncf.py:186) input_step_eval_neg_gen
:::NVLOGv0.1.0 ncf 1603861910.924700260 (ncf.py:194) run_start
:::NVLOGv0.1.0 ncf 1603861911.073457718 (ncf.py:201) preproc_hp_num_eval: 100
:::NVLOGv0.1.0 ncf 1603861911.074674845 (ncf.py:207) input_size: 19861770
:::NVLOGv0.1.0 ncf 1603861911.099672079 (ncf.py:216) input_batch_size: 1200000
:::NVLOGv0.1.0 ncf 1603861911.099897861 (ncf.py:217) input_order
:::NVLOGv0.1.0 ncf 1603861911.100259066 (/workspace/examples/ncf/neumf.py:54) model_hp_mf_dim: 64
:::NVLOGv0.1.0 ncf 1603861911.328058004 (/workspace/examples/ncf/neumf.py:62) model_hp_mlp_layer_sizes: [256, 256, 128, 64]
Selected optimization level O2:  FP16 training with FP32 batchnorm and FP32 master weights.

Defaults for this optimization level are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : True
master_weights         : True
loss_scale             : dynamic
Processing user overrides (additional kwargs that are not None)...
After processing overrides, optimization options are:
enabled                : True
opt_level              : O2
cast_model_type        : torch.float16
patch_torch_functions  : False
keep_batchnorm_fp32    : False
master_weights         : True
loss_scale             : dynamic
NeuMF(
  (mf_user_embed): Embedding(138493, 64)
  (mf_item_embed): Embedding(26744, 64)
  (mlp_user_embed): Embedding(138493, 128)
  (mlp_item_embed): Embedding(26744, 128)
  (mlp): ModuleList(
    (0): Linear(in_features=256, out_features=256, bias=True)
    (1): Linear(in_features=256, out_features=128, bias=True)
    (2): Linear(in_features=128, out_features=64, bias=True)
  )
  (final): Linear(in_features=128, out_features=1, bias=True)
)
31832577 parameters
:::NVLOGv0.1.0 ncf 1603861911.595915318 (ncf.py:246) opt_learning_rate: 0.0045
:::NVLOGv0.1.0 ncf 1603861911.596179724 (ncf.py:247) opt_name: "Adam"
:::NVLOGv0.1.0 ncf 1603861911.596397877 (ncf.py:248) opt_hp_Adam_beta1: 0.25
:::NVLOGv0.1.0 ncf 1603861911.596611261 (ncf.py:249) opt_hp_Adam_beta2: 0.5
:::NVLOGv0.1.0 ncf 1603861911.596828461 (ncf.py:250) opt_hp_Adam_epsilon: 1e-08
:::NVLOGv0.1.0 ncf 1603861911.597044945 (ncf.py:251) model_hp_loss_fn: "binary_cross_entropy"
:::NVLOGv0.1.0 ncf 1603861911.597252607 (ncf.py:279) train_loop
:::NVLOGv0.1.0 ncf 1603861911.597468853 (ncf.py:282) train_epoch_start: 0
:::NVLOGv0.1.0 ncf 1603861911.597685575 (ncf.py:283) input_hp_num_neg: 4
:::NVLOGv0.1.0 ncf 1603861911.597892284 (ncf.py:284) input_step_train_neg_gen
:::NVLOGv0.1.0 ncf 1603861917.104950190 (ncf.py:318) train_throughput: 18034728.960925747
:::NVLOGv0.1.0 ncf 1603861917.105417967 (ncf.py:319) train_epoch_stop: 0
:::NVLOGv0.1.0 ncf 1603861917.105912209 (ncf.py:320) eval_start: 0
ncf.py:146: UserWarning: indexing with dtype torch.uint8 is now deprecated, please use a dtype torch.bool instead. (Triggered internally at  ../aten/src/ATen/native/IndexingUtils.h:25.)
  temp[dup_mask] = -1
ncf.py:153: UserWarning: This overload of nonzero is deprecated:
	nonzero(Tensor input, *, Tensor out)
Consider using one of the following signatures instead:
	nonzero(Tensor input, *, bool as_tuple) (Triggered internally at  ../torch/csrc/utils/python_arg_parser.cpp:882.)
  ndcg = (math.log(2) / (torch.nonzero(ifzero)[:,1].view(-1).to(torch.float)+2).log_()).sum()
:::NVLOGv0.1.0 ncf 1603861917.369261742 (ncf.py:155) eval_size: {"epoch": 0, "value": 13987793}
:::NVLOGv0.1.0 ncf 1603861917.369539261 (ncf.py:156) eval_hp_num_users: 138493
:::NVLOGv0.1.0 ncf 1603861917.369806051 (ncf.py:157) eval_hp_num_neg: 100
Epoch 0: HR@10 = 0.8756, NDCG@10 = 0.6072, train_time = 5.51, val_time = 0.27
:::NVLOGv0.1.0 ncf 1603861917.370156527 (ncf.py:333) eval_accuracy: {"epoch": 0, "value": 0.8755604976424801}
:::NVLOGv0.1.0 ncf 1603861917.370383978 (ncf.py:334) eval_target: 1.0
:::NVLOGv0.1.0 ncf 1603861917.370639324 (ncf.py:335) eval_stop: 0
:::NVLOGv0.1.0 ncf 1603861917.370875835 (ncf.py:340) eval_throughput: 52695968.14215835
New best hr! Saving the model to:  /data/checkpoints/model.pth
:::NVLOGv0.1.0 ncf 1603861917.644968033 (ncf.py:282) train_epoch_start: 1
:::NVLOGv0.1.0 ncf 1603861917.645388842 (ncf.py:283) input_hp_num_neg: 4
:::NVLOGv0.1.0 ncf 1603861917.645800352 (ncf.py:284) input_step_train_neg_gen
Traceback (most recent call last):
  File "ncf.py", line 366, in <module>
    main()
  File "ncf.py", line 288, in main
    epoch_users, epoch_items, epoch_label = dataloading.prepare_epoch_train_data(train_ratings, nb_items, args)
  File "/workspace/examples/ncf/dataloading.py", line 100, in prepare_epoch_train_data
    epoch_indices = torch.randperm(train_users_end - train_users_begin, device='cuda:{}'.format(args.local_rank))
RuntimeError: CUDA out of memory. Tried to allocate 1.48 GiB (GPU 0; 9.78 GiB total capacity; 6.33 GiB already allocated; 853.69 MiB free; 7.98 GiB reserved in total by PyTorch)
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/opt/conda/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/launch.py", line 261, in <module>
    main()
  File "/opt/conda/lib/python3.6/site-packages/torch/distributed/launch.py", line 257, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/opt/conda/bin/python', '-u', 'ncf.py', '--local_rank=0', '--data', '/data/ncf/cache/ml-20m', '--epochs', '2', '--batch_size', '1200000', '--opt_level', 'O2']' returned non-zero exit status 1.
DONE!
